packages/components/src/index.ts

```diff
@@ -116,15 +188,45 @@ export { default as __experimentalNavigationGroup } from './navigation/group';
 export { default as __experimentalNavigationItem } from './navigation/item';
 export { default as __experimentalNavigationMenu } from './navigation/menu';
 export {
+	/**
+	 * @deprecated Import `NavigatorProvider` instead.
+	 */
 	NavigatorProvider as __experimentalNavigatorProvider,
+	/**
+	 * @deprecated Import `NavigatorScreen` instead.
+	 */
 	NavigatorScreen as __experimentalNavigatorScreen,
+	/**
+	 * @deprecated Import `NavigatorButton` instead.
+	 */
 	NavigatorButton as __experimentalNavigatorButton,
+	/**
+	 * @deprecated Import `NavigatorBackButton` instead.
+	 */
 	NavigatorBackButton as __experimentalNavigatorBackButton,
+	/**
+	 * @deprecated Import `NavigatorToParentButton` instead.
+	 */
 	NavigatorToParentButton as __experimentalNavigatorToParentButton,
```

Comment:
@mirka, Based on your comment here https://github.com/WordPress/gutenberg/pull/60837/files/ce4f6386c86c9f4b3a119888aa1ad1326f373ef4..51d389b4cb73811b2eed88aab931abf75f5486ed#diff-bef15a411502a2e04eb09ba2277b51079c8883c8cafd650fa7d1bb8873d03fb9, I'm wondering how we should approach this. It seems we should only deprecate the components and leave any functions as is? Does that mean I should have left the hook aliased to its `__experimental` alias?

By the way, I think this might call for moving those components as subcomponents (e.g `Navigator.Provider`, `Navigator.Screen`, `Navigator.BackButton`, etc)?

Reply:
> It seems we should only deprecate the components and leave any helper functions as is?

Yes, I think so. Hope we don't find anymore helper functions though 😅 `use*` hooks should be fine.

> By the way, I think this might call for moving those components as subcomponents (e.g `Navigator.Provider`, `Navigator.Screen`, `Navigator.BackButton`, etc)?

Good point, this would be a great opportunity to do that 😱 It will be a little bit more work for our consumers to update their code from the `__experimental`, but we do indeed want to move to dot notation namespacing (`Navigator.Provider`) over "flat" namespacing (`NavigatorProvider`). Could you try it for one of the components so we can all check that there are no implications we're failing to consider?

Reply:
> but we do indeed want to move to dot notation namespacing (Navigator.Provider) over "flat" namespacing (NavigatorProvider). Could you try it for one of the components so we can all check that there are no implications we're failing to consider?

Sure! Wil do 👍🏻

Reply:
I started experimenting with it [here](https://github.com/WordPress/gutenberg/pull/60926); it'd allow consumers to import only the root component (the provider), and then the rest would come built-in as properties of the function component. I think this makes sense, as these components are building blocks for the same thing; they can't be used in isolation, by themselves _(though some might be omitted, I guess, like one of the buttons; see the comment about tree-shaking below)_. I still need to solve some TS errors, though. 

Assuming TS errors are solved, there are a few other things to discuss, like the naming of the root component and if doing this could negatively affect tree shaking the way it's currently setup (I don't know enough about the webpack setup to judge that, yet).